Game/MyProject/Game/RoadGenerator/RoadGeneratorControler.hpp

```diff
@@ -23,7 +23,15 @@ class RoadGeneratorControler : std::enable_shared_from_this<RoadGeneratorControl
     const std::vector<std::pair<IvVector3,IvVector3>>& getMarginPoints() const { return rMarginPoints;};
     const int getEditIndex() const {return editIndex;};
     
+    void previousPoint();
+    void nextPoint();
+    void addPoint();
+    void removePoint();
+    void generateTrack();
+    
 private:
+    bool needsUpdate=false;
+    bool hasUpdate=false;
```

Comment:
What's the point of having both *needsUpdate* and *hasUpdate*?
Please find better logic for this, it seems redundant.